xclim/indices/_agro.py

```diff
@@ -589,6 +595,221 @@ def water_budget(
     return out
 
 
+@declare_units(
+    pr="[precipitation]",
+    pr_cal="[precipitation]",
+)
+def standardized_precipitation_index(
+    pr: xarray.DataArray,
+    pr_cal: xarray.DataArray,
+    freq: str = "MS",
+    window: int = 1,
+    dist: str = "gamma",
+    method: str = "APP",
+) -> xarray.DataArray:
+    r"""Standardized Precipitation Index (SPI).
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+      Daily precipitation.
+    pr_cal : xarray.DataArray
+      Daily precipitation used for calibration. Usually this is a temporal subset of `pr` over some reference period.
+    freq : str
+      Resampling frequency. A monthly or daily frequency is expected.
+    window : int
+      Averaging window length relative to the resampling frequency. For example, if `freq="MS"`, i.e. a monthly resampling, the window
+      is an integer number of months.
+    dist : {'gamma'}
+      Name of the univariate distribution, only `gamma` is currently implemented
+      (see :py:mod:`scipy.stats`).
+    method : {'APP', 'ML'}
+      Name of the fitting method, such as `ML` (maximum likelihood), `APP` (approximate). The approximate method
+      uses a deterministic function that doesn't involve any optimization.
+
+    Returns
+    -------
+    xarray.DataArray,
+      Standardized Precipitation Index.
+
+
+    Notes
+    -----
+    The length `N` of the N-month SPI is determined by choosing the `window = N`. Supported statistical distributions are: ["gamma"]
+
+
+    Example
+    -------
+    Computing SPI-3 months using a gamma distribution for the fit
+
+    .. code-block:: python
+
+        import xclim.indices as xci
+        import xarray as xr
+
+        ds = xr.open_dataset(filename)
+        pr = ds.pr
+        pr_cal = pr.sel(time=slice(calibration_start_date, calibration_end_date))
+        spi_3 = xci.standardized_precipitation_index(
+            pr, pr_cal, freq="MS", window=3, dist="gamma", method="ML"
+        )
+
+
+    References
+    ----------
+    McKee, Thomas B., Nolan J. Doesken, and John Kleist. "The relationship of drought frequency and duration to time scales." In Proceedings of the 8th Conference on Applied Climatology, vol. 17, no. 22, pp. 179-183. 1993.
+    """
+    # "WPM" method doesn't seem to work for gamma or pearson3
+    dist_and_methods = {"gamma": ["ML", "APP"]}
+    if dist not in dist_and_methods.keys():
+        raise NotImplementedError(f"The distribution `{dist}` is not supported.")
+    elif method not in dist_and_methods[dist]:
+        raise NotImplementedError(
+            f"The method `{method}` is not supported for distribution `{dist}`."
+        )
+
+    # calibration period
+    cal_period = [pr_cal.time.values[0], pr_cal.time.values[-1]]
```

Comment:
```suggestion
    cal_period = pr_cal.time[[0, -1]].dt.strftime('%Y-%m-%dT%H:%M:%S').values.tolist()
```
List indexing to be fancy, but most importantly, this formats the dates into a string. NetCDFs do not support datetime objects in their attributes, and this string format is very common (ISO 8601, without the timezone info).